src/client/actions/ChannelDelete.js

```diff
@@ -17,7 +18,14 @@ class ChannelDeleteAction extends Action {
     } else {
       channel = this.deleted.get(data.id) || null;
     }
-    if (channel) channel.deleted = true;
+    if (channel) {
+      if (channel.messages && !(channel instanceof DMChannel)) {
```

Comment:
```suggestion
      if (channel.messages && channel.type !== 'dm') {
```

Small change but then the `require` is unnecessary